opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -580,880 +354,418 @@ class Init : private boost::noncopyable
-                // W{O,G,W}{I,P}R
-                addWell(std::unique_ptr <WellReport>(
-                            new WellRate(*this,
-                                         eclipseState,
-                                         wells[wellIdx],
-                                         uses,
-                                         ertPhaseIdx,
-                                         wellType,
-                                         deckUnitType)));
-                // W{O,G,W}{I,P}T
-                addWell(std::unique_ptr <WellReport>(
-                            new WellTotal(*this,
-                                          eclipseState,
-                                          wells[wellIdx],
-                                          uses,
-                                          ertPhaseIdx,
-                                          wellType,
-                                          deckUnitType)));
-            }
-        }
-    }
+inline std::string uppercase( std::string x ) {
+    std::transform( x.begin(), x.end(), x.begin(),
+        []( char c ) { return std::toupper( c ); } );
 
-    // Add BHP monitors
-    for (int wellIdx = 0; wellIdx != numWells; ++wellIdx) {
-        // In the call below: uses, phase and the well type arguments
-        // are not used, except to set up an index that stores the
-        // well indirectly. For details see the implementation of the
-        // WellReport constructor, and the method
-        // WellReport::bhp().
-        BlackoilPhases::PhaseIndex ertPhaseIdx = BlackoilPhases::Liquid;
-        if (!uses.phase_used[BlackoilPhases::Liquid]) {
-            ertPhaseIdx = BlackoilPhases::Vapour;
-        }
-        addWell(std::unique_ptr <WellReport>(
-                    new WellBhp(*this,
-                                eclipseState,
-                                wells[wellIdx],
-                                uses,
-                                ertPhaseIdx,
-                                WELL_TYPES[0],
-                                deckUnitType)));
-    }
+    return x;
 }
-} // end namespace EclipseWriterDetails
-
 
-
-/**
- * Convert opm-core WellType and InjectorType to eclipse welltype
- */
-int EclipseWriter::eclipseWellTypeMask(WellType wellType, WellInjector::TypeEnum injectorType)
-{
-  int ert_well_type = IWEL_UNDOCUMENTED_ZERO;
-
-  if (PRODUCER == wellType) {
-      ert_well_type = IWEL_PRODUCER;
-  } else if (INJECTOR == wellType) {
-      switch (injectorType) {
-        case WellInjector::WATER:
-          ert_well_type = IWEL_WATER_INJECTOR;
-          break;
-        case WellInjector::GAS:
-          ert_well_type = IWEL_GAS_INJECTOR;
-          break;
-        case WellInjector::OIL :
-          ert_well_type = IWEL_OIL_INJECTOR;
-          break;
-        default:
-          ert_well_type = IWEL_UNDOCUMENTED_ZERO;
-      }
-  }
-
-  return ert_well_type;
 }
 
+class EclipseWriter::Impl {
```

Comment:
What, the private implementation? It saves us exposing all the internal RFT/Restart/Init etc. to users, which is nice.

If you're talking about the name itself I don't have that strong an opinion (since it's isolated anyway), but I figured it was better than my original idea of "E" or "EW".